contracts/pools/aeth/StableSwapAETH.vy

```diff
@@ -0,0 +1,857 @@
+    # dx and dy in c-units
+    rates: uint256[N_COINS] = self._stored_rates()
+    xp: uint256[N_COINS] = self._xp(rates)
+
+    x: uint256 = xp[i] + dx * rates[i] / PRECISION
+    y: uint256 = self.get_y(i, j, x, xp)
+    dy: uint256 = xp[j] - y
+    _fee: uint256 = self.fee * dy / FEE_DENOMINATOR
+    return (dy - _fee) * PRECISION / rates[j]
+
+
+@view
+@external
+def get_dx(i: int128, j: int128, dy: uint256) -> uint256:
+    # dx and dy in c-units
+    rates: uint256[N_COINS] = self._stored_rates()
+    xp: uint256[N_COINS] = self._xp(rates)
+
+    y: uint256 = xp[j] - (dy * FEE_DENOMINATOR / (FEE_DENOMINATOR - self.fee)) * rates[j] / PRECISION
+    x: uint256 = self.get_y(j, i, y, xp)
+    dx: uint256 = (x - xp[i]) * PRECISION / rates[i]
+    return dx
+
+
+@payable
+@external
+@nonreentrant('lock')
+def exchange(i: int128, j: int128, dx: uint256, min_dy: uint256) -> uint256:
+    """
+    @notice Perform an exchange between two coins
+    @dev Index values can be found via the `coins` public getter method
+    @param i Index value for the coin to send
+    @param j Index valie of the coin to recieve
+    @param dx Amount of `i` being exchanged
+    @param min_dy Minimum amount of `j` to receive
+    @return Actual amount of `j` received
+    """
+    assert not self.is_killed # dev: is killed
+
+    rates: uint256[N_COINS] = self._stored_rates()
+
+    xp: uint256[N_COINS] = self._xp(rates)
+    x: uint256 = xp[i] + dx * rates[i] / PRECISION
+    y: uint256 = self.get_y(i, j, x, xp)
+    dy: uint256 = xp[j] - y - 1  # -1 just in case there were some rounding errors
+    dy_fee: uint256 = dy * self.fee / FEE_DENOMINATOR
+    dy_admin_fee: uint256 = dy_fee * self.admin_fee / FEE_DENOMINATOR
+
+    self.balances[i] = x * PRECISION / rates[i]
+    self.balances[j] = (y + (dy_fee - dy_admin_fee)) * PRECISION / rates[j]
+
+    dy = (dy - dy_fee) * PRECISION / rates[j]
+    assert dy >= min_dy, "Exchange resulted in fewer coins than expected"
+
+    coin: address = self.coins[1]
+    if i == 0:
+        assert msg.value == dx
+        assert ERC20(coin).transfer(msg.sender, dy)
+    else:
+        assert msg.value == 0
+        assert ERC20(coin).transferFrom(msg.sender, self, dx)
+        raw_call(msg.sender, b"", value=dy)
+
+    log TokenExchange(msg.sender, i, dx, j, dy)
+
+    return dy
+
+
+@external
+@nonreentrant('lock')
+def remove_liquidity(_amount: uint256, min_amounts: uint256[N_COINS]) -> uint256[N_COINS]:
+    """
+    @notice Withdraw coins from the pool
+    @dev Withdrawal amounts are based on current deposit ratios
+    @param _amount Quantity of LP tokens to burn in the withdrawal
+    @param min_amounts Minimum amounts of underlying coins to receive
+    @return List of amounts of coins that were withdrawn
+    """
+    _lp_token: address = self.lp_token
+    total_supply: uint256 = ERC20(_lp_token).totalSupply()
+    amounts: uint256[N_COINS] = empty(uint256[N_COINS])
+    fees: uint256[N_COINS] = empty(uint256[N_COINS])
```

Comment:
there is no need to declare `fees`, the call to `empty` can be placed within the event that's emitted on line 520